common/log.c

```diff
@@ -532,17 +533,20 @@ internal_log_is_enabled_for_level(const enum logLevels log_level,
 
     /* Is there at least one log destination which will accept the message based on the log level? */
     return (g_staticLogConfig->fd >= 0
-            && (override_destination_level || log_level <= g_staticLogConfig->log_level))
+               && ((override_destination_level && log_level <= override_log_level)
+                  || (!override_destination_level && log_level <= g_staticLogConfig->log_level)))
            || (g_staticLogConfig->enable_syslog
-               && (override_destination_level || log_level <= g_staticLogConfig->syslog_level))
+               && ((override_destination_level && log_level <= override_log_level)
+                  || (!override_destination_level && log_level <= g_staticLogConfig->syslog_level)))
            || (g_staticLogConfig->enable_console
-               && (override_destination_level || log_level <= g_staticLogConfig->console_level));
+               && ((override_destination_level && log_level <= override_log_level)
+                  || (!override_destination_level && log_level <= g_staticLogConfig->console_level)));
```

Comment:
I'm a little concerned with this mega-conditional. It's quite hard to parse mentally (at least it is for me), and there's some repetition in it. Maybe split it up a bit? Here's a suggestion:-

```c
{
    bool_t result;
    /* Is log initialized? */
    if (g_staticLogConfig == NULL)
    {
        result = 0;
    }
    else if (override_destination_level)
    {
        /* Override in place - if the message should be logged, is any
         * destination enabled? */
        if (log_level > override_log_level)
        {
            result = 0;
        }
        else
        {
            result =
                g_staticLogConfig->fd >= 0 ||
                g_staticLogConfig->enable_syslog ||
                g_staticLogConfig->enable_console;
        }
    }
    else
    {
        /* Is there at least one log destination which will accept the
         * message based on the log level? */
        result =
            (g_staticLogConfig->fd >= 0 &&
             log_level <= g_staticLogConfig->log_level) ||
            (g_staticLogConfig->enable_syslog &&
             log_level <= g_staticLogConfig->syslog_level) ||
            (g_staticLogConfig->enable_console &&
             log_level <= g_staticLogConfig->console_level);
    }

    return result;
}
```

Reply:
cleaned up to be easier to understand.